SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -2215,6 +2216,74 @@ void CEulerSolver::Source_Residual(CGeometry *geometry, CSolver **solver_contain
+    /*!
+     * \brief Vorticity Confinement (VC) technique to counter the numerical
+     * diffusion offered by the numerical scheme
+     * \author: Y Chandukrishna, Josy P Pullockara, T N Venkatesh.
+     * Computational and Theoretical Fluid Dynamics division,
+     * CSIR-National Aerospace Laboratories (NAL), Bangalore.
+     * Academy of Scientific and Innovative Research (AcSIR), Ghaziabad.
+     * First release date : 23 December 2022
+     * modified on:
+     *
+     * VC technique introduces an additional term to the N-S equations that
+     * counters the numerical difusion offerd by the numerical schemes. The
+     * additional term is introduced as a source term. VC technique requires an
+     * input confinement parameter that controls the magnitude of the source
+     * term. A suitable Confinement parameter is problem, scheme and grid
+     * dependant.
+     *
+     * Required changes in config file -
+     * VORTICITY_CONFINEMENT = YES
+     * CONFINEMENT_PARAMETER = <0.0(default)>
+     * Currently have support for Viscous, Inviscid cases for both 2D and 3D
+     * problems. Can be used along with other source terms also.
+     *
+     * The current implementation follows,
+     * R. Loehner and C. Yang, Vorticity confinement on unstructured grids, 2002.
+     * N. Butsuntorn and A. Jameson, Time Spectral Method for Rotorcraft Flow, 2008.
+     */
+
+    CNumerics* second_numerics = numerics_container[SOURCE_SECOND_TERM + omp_get_thread_num()*MAX_TERMS];
+
+    su2double VolumeSum_global = 0.0;
+    su2double VolumeSum_local = 0.0;
+    for (iPoint = 0; iPoint < nPoint; iPoint++) {
+      // for calculating average cell volume
+      VolumeSum_local += geometry->nodes->GetVolume(iPoint);
+
+      // set vorticity magnitude as auxilliary variable
+      const su2double VorticityMag = max(GeometryToolbox::Norm(3, nodes->GetVorticity(iPoint)), 1e-12);
+      nodes->SetAuxVar(iPoint, 0, VorticityMag);
+    }
+    SU2_MPI::Allreduce(&VolumeSum_local, &VolumeSum_global, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+    const unsigned long nPointLocal = geometry->GetnPoint();
+    unsigned long nPointGlobal = 0;
+    SU2_MPI::Allreduce(&nPointLocal, &nPointGlobal, 1, MPI_UNSIGNED_LONG, MPI_SUM, MPI_COMM_WORLD);
+    const su2double AvgVolume = VolumeSum_global / nPointGlobal;
+    second_numerics->SetAvgVolume_VC(AvgVolume);
+
+    SetAuxVar_Gradient_GG(geometry, config);
+
+    SU2_OMP_FOR_DYN(omp_chunk_size)
+    for (iPoint = 0; iPoint < nPointDomain; iPoint++) {
+      second_numerics->SetConservative(nodes->GetSolution(iPoint), nodes->GetSolution(iPoint));
+      second_numerics->SetPrimitive(nodes->GetPrimitive(iPoint), nodes->GetPrimitive(iPoint));
+      second_numerics->SetVorticity(nodes->GetVorticity(iPoint), nodes->GetVorticity(iPoint));
+      second_numerics->SetAuxVarGrad(nodes->GetAuxVarGradient(iPoint), nullptr);
+      second_numerics->SetWallDistance_Numerics_VC(geometry->nodes->GetWall_Distance(iPoint));
```

Comment:
After the other suggestions it should look like this.
```suggestion
      second_numerics->SetPrimitive(nodes->GetPrimitive(iPoint), nullptr);
      second_numerics->SetVorticity(nodes->GetVorticity(iPoint), nullptr);
      second_numerics->SetAuxVarGrad(nodes->GetAuxVarGradient(iPoint), nullptr);
      second_numerics->SetDistance(geometry->nodes->GetWall_Distance(iPoint), 0);
```